blenderproc/python/types/URDFUtility.py

```diff
@@ -0,0 +1,276 @@
+from typing import Union, List
+import numpy as np
+from mathutils import Matrix
+
+import bpy
+
+from blenderproc.python.types.EntityUtility import Entity
+from blenderproc.python.types.MeshObjectUtility import MeshObject
+from blenderproc.python.types.ArmatureUtility import Armature
+
+
+class Inertial(Entity):
+    def __init__(self, bpy_object: bpy.types.Object):
+        super().__init__(bpy_object=bpy_object)
+
+        object.__setattr__(self, "inertia", None)
+        object.__setattr__(self, "mass", None)
+        object.__setattr__(self, "origin", None)
+
+    def set_inertia(self, inertia: np.ndarray):
+        """ Sets inertia value.
+
+        :param inertia: 3x3 symmetric rotational inertia matrix.
+        """
+        assert inertia.shape == (3, 3)
+        object.__setattr__(self, "inertia", inertia)
+
+    def get_inertia(self) -> np.ndarray:
+        """ Returns the inertia.
+
+        :return: The inertia matrix.
+        """
+        return self.inertia
+
+    def set_mass(self, mass: float):
+        """ Sets the mass.
+
+        :param mass: Mass of the link in kilograms.
+        """
+        object.__setattr__(self, "mass", mass)
+
+    def get_mass(self) -> float:
+        """ Returns the mass of the link.
+
+        :return: The mass.
+        """
+        return self.mass
+
+    def set_origin(self, origin: Union[np.ndarray, Matrix]):
+        """ Sets the origin and the world matrix of the inertia.
+
+        :param origin: 4x4 matrix of the inertials relative to the link frame.
+        """
+        object.__setattr__(self, "origin", Matrix(origin))
+        self.blender_obj.matrix_world = Matrix(origin)
+
+    def get_origin(self) -> Matrix:
+        """ Returns the origin of the inertia.
+
+        :return: The pose relative to the link frame.
+        """
+        return self.origin
+
+
+class Link(Armature):
+    def __init__(self, bpy_object):
+        super().__init__(bpy_object=bpy_object)
+
+        object.__setattr__(self, 'visuals', [])
```

Comment:
As using `__setattr__` is quite ugly, why do we not overwrite `__setattr__` in this class and then allow directly setting the attributes `'visuals'`, `'inertial'`... @themasterlink

Reply:
The advantage is that usually everything should be stored inside of a `blenderobj` as we can then use the key frame mechanics to control it. Values outside of them can not be controlled :( 

I would try to avoid using attributes which are not stored in the corresponding object. 

Can we store those two attributes inside of a `blenderobj`?

Reply:
But which attributes would you like to keyframe here? The attributes are mostly references to other objects.

I mean, using attributes here is not optimal, but I think okay in this place. What I criticize is more is that we use `object.__setattr__(self, ...)` instead of `self.inertial = ....`

Reply:
I guess now there are many more `__setattr__()` some of which require key frame mechanics. Maybe we can talk about the best solution on Friday.